VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/hearing-rules/hearing-rules.component.ts

```diff
@@ -1,29 +1,42 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Router, ActivatedRoute } from '@angular/router';
 import { pageUrls } from 'src/app/shared/page-url.constants';
 import { ParticipantStatusBaseDirective } from 'src/app/on-the-day/models/participant-status-base';
 import { ParticipantStatusUpdateService } from 'src/app/services/participant-status-update.service';
 import { Logger } from 'src/app/services/logging/logger-base';
+import { Subscription } from 'rxjs';
+import { HearingVenueFlagsService } from 'src/app/services/hearing-venue-flags.service';
 
 @Component({
     selector: 'app-hearing-rules',
     templateUrl: './hearing-rules.component.html',
     styleUrls: ['./hearing-rules.component.scss']
 })
-export class HearingRulesComponent extends ParticipantStatusBaseDirective implements OnInit {
+export class HearingRulesComponent extends ParticipantStatusBaseDirective implements OnInit, OnDestroy {
     conferenceId: string;
+    hearingVenueIsInScotland = false;
+    hearingVenueFlagsServiceSubscription$: Subscription;
 
     constructor(
         private router: Router,
         protected route: ActivatedRoute,
         protected participantStatusUpdateService: ParticipantStatusUpdateService,
-        protected logger: Logger
+        protected logger: Logger,
+        private hearingVenueFlagsService: HearingVenueFlagsService
     ) {
         super(participantStatusUpdateService, logger);
     }
 
     ngOnInit() {
         this.conferenceId = this.route.snapshot.paramMap.get('conferenceId');
+
+        this.hearingVenueFlagsServiceSubscription$ = this.hearingVenueFlagsService.hearingVenueIsScottish$.subscribe(
+            value => (this.hearingVenueIsInScotland = value)
```

Comment:
This would actually be a good place to use the async pipe which should be preferred for these cases then we don't need to handle the subscription and disposing it.

https://angular.io/api/common/AsyncPipe

Reply:
how would I use the async pipe in the template, I am using the flag hearingVenueIsInScotland in the template and not the hearingVenueFlagsServiceSubscription$

Reply:
that's why I suggested creating a directive that supports inputs. I'm not going to stop you on this, just something to think about